pkgs/development/ocaml-modules/cstruct/default.nix

```diff
@@ -1,20 +1,40 @@
-{stdenv, writeText, fetchurl, ocaml, ocplib-endian, sexplib_p4, findlib,
- async_p4 ? null, lwt ? null, camlp4}:
+{stdenv, buildOcaml, fetchFromGitHub, writeText,
+ ocaml, ocplib-endian, sexplib_p4, findlib, ounit, camlp4,
```

Comment:
to jump in here, cstruct 2.3.0 has dropped camlp4 support -- thus this is no longer a dependency.

from the opam file (https://github.com/ocaml/opam-repository/blob/master/packages/cstruct/cstruct.2.3.0/opam):
```
depends: [
  "ocamlfind" {build}
  "ounit"     {test}
  "ocplib-endian"
  "sexplib"
  "base-bytes"
]
depopts: [
  "ppx_tools"
  "async"
  "lwt"
]
```

not sure what your `sexplib_p4` is (I suspect a sexplib with camlp4 support), but cstruct does no longer need this, a sexplib without p4 support is fine!